client/dashboard/domains/domain-transfer/index.tsx

```diff
@@ -0,0 +1,292 @@
+					description={ description }
+					actions={
+						<RouterLinkButton
+							variant="secondary"
+							size="compact"
+							to={ `/domains/${ domain.domain }/transfer/other-user` }
+						>
+							{ __( 'Continue' ) }
+						</RouterLinkButton>
+					}
+				/>
+			);
+		} else if (
+			! RESTRICTED_TRANSFER_TLDS.includes( getTopLevelOfTld( domain.domain ) ) &&
+			domain.can_transfer_to_other_site
+		) {
+			actions.push(
+				<ActionList.ActionItem
+					key="transfer-to-any-user"
+					title={ __( 'To another WordPress.com user' ) }
+					description={ __( 'Transfer this domain to another WordPress.com user' ) }
+					actions={
+						<RouterLinkButton
+							variant="secondary"
+							size="compact"
+							to={ `/domains/${ domain.domain }/transfer/any-user` }
+						>
+							{ __( 'Continue' ) }
+						</RouterLinkButton>
+					}
+				/>
+			);
+		}
+		if ( domain.can_transfer_to_other_site ) {
+			const description =
+				domain.subtype.id === DomainSubtype.DOMAIN_CONNECTION
+					? __( 'Transfer this domain connection to any site you are an administrator on.' )
+					: __( 'Transfer this domain to any site you are an administrator on.' );
+			actions.push(
+				<ActionList.ActionItem
+					key="transfer-to-another-site"
+					title={ __( 'To another WordPress.com site' ) }
+					description={ description }
+					actions={
+						<RouterLinkButton
+							variant="secondary"
+							size="compact"
+							to={ `/domains/${ domain.domain }/transfer/other-site` }
+						>
+							{ __( 'Continue' ) }
+						</RouterLinkButton>
+					}
+				/>
+			);
+		}
+		if ( actions.length > 0 ) {
+			return <ActionList>{ actions }</ActionList>;
+		}
+	};
+
+	const renderTransferMessage = () => {
+		const registrationDate = new Date( domain.registration_date );
+		const today = new Date();
+		const registrationDatePlus60Days = new Date(
+			registrationDate.getTime() + TRANSFER_LOCK_GRACE_PERIOD_MS
+		);
+
+		let message: string | React.ReactElement = __( 'This domain cannot be locked.' );
+
+		if ( domain.domain_locking_available ) {
+			message = __(
+				'We recommend leaving the transfer lock on, unless you want to transfer your domain to another provider.'
+			);
+		}
+
+		const supportLink =
+			today > registrationDatePlus60Days
+				? 'domain-designated-agent'
+				: 'transfer-domain-registration';
+
+		if ( domain.transfer_away_eligible_at ) {
+			const transferAwayEligibleAt = formatDate(
+				new Date( domain.transfer_away_eligible_at ),
+				locale
+			);
+
+			message = createInterpolateElement(
+				// translators: <date> is a date string, <link> is a link to the support page
+				__( 'You can unlock this domain after <date/>. <link>Why is my domain locked?</link>' ),
+				{
+					date: <>{ transferAwayEligibleAt }</>,
+					link: <InlineSupportLink supportContext={ supportLink } />,
+				}
+			);
+		}
+
+		return <Text variant="muted">{ message }</Text>;
+	};
+
+	const renderTransferLock = () => {
+		const disabled = Boolean(
```

Comment:
Do we need Boolean here?

Reply:
yep, `transfer_away_eligible_at` is a string. I'm not a big fan of !! syntax but honestly I don't have strong opinion 😅